### ISSUE TITLE
award.identifier.present check

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-checks#101]
## Description

Check if an award identifier is present

## Priority

- FAIR: Optional

## Issues
- this check is dependent on EML 2.2 (i.e. <award_number>, <awardUrl>?)

**Comment by mbjones** [comment]
This check would work for EML 2.2, for DataCite, for schema.org, and for older versions of EML if we search in the `funding` element for the award number.

**Comment by tedhabermann** [comment]
Make sure we are getting the award identifier and not the funder identifier...

**Comment by gothub** [comment]
Any recommendations on conventions for parsing the `<funding>` element for `funder` and `award identifier` would help. Here are a few examples from DataONE:
```
Blandy.352.1
<funding><para>National Science Foundation DBI 1156796</para></funding>

Blandy.8.9
<funding><para>PISCO is funded by the David and Lucile Packard Foundation and the Gordon and Betty Moore Foundation.</para></funding>

MontanaIOE.12.7
<funding><para>NSF Awards EF-1029808, EF-1246537, EF-1138160, EF-0752017</para></funding>

NEON.D14.JORN.DP1.00001.1
<funding><para>PPI-INPA 015/122</para> </funding>

PPBioAmOc.211.2
<funding> <para>1. CNPq - Universal 473474/2008-5/ 2. CNPq/ - CT Amazônia 575637/2008-0/ 3. CAPES - bolsa de doutorado (concedida a Cíntia Gomes de Freitas).</para></funding>

PPBioMA.28.8
<funding><para>CNPq  XXXXX, XXXXXX; FAPEG XXXXXX</para> </funding>

arctic-data.14757.2
<funding><para>1203418</para><para>1203451</para>

autogen.2016032221524829393.9
<funding> <para>Exxon Valdez Oil Spill Trsutee Council project numbers 00163-K, 99163-K, 98163-K, 97163-K, 99163-K, 95163-K</para></funding>

autogen.2016071911451366602.1
<funding>Western Virginia Water Authority</funding>

doi:10.18739/A21V5BD3Z
<funding><para>OPP-9732051</para> <para>OPP-0352958</para></funding>

doi:10.6085/AA/SH70XX_070MTV2050R00_20180622.50.1
<funding><para>NSF (LTER) MCR I #OCE-0417412, MCR II #OCE-1026851, and MCR IIb #OCE-1236905</para></funding>

doi:10.6085/AA/FCKX00_XXXITV2XMMR02_20180417.50.1
<funding><para>NOAA NA11NMF4570218</para><para>EPA FP91767501</para><para>NSF GK-12 DGE-0840804</para></funding>
```

**Comment by gothub** [comment]
@mbjones There are 3 different checks for FAIR concerning funding information: `award.funder.present`, `award funderIdentifier.present`, `award.identifier.present`. For EML < 2.2, do we only look for the `award identifier` and not funder info, i.e. only this check `award.identifier.present` is looking int he EML < 2.2 field `/eml/dataset/project/funding` element?

**Comment by mbjones** [comment]
Yeah, for eml 2.2.0 we can find all 3 and check presence just based on the structured fields. 

For earlier versions of EML, the `funding` text field will likely have a mix of funder names/acronyms and award numbers in various formats and mixed in with other text.  Given common patterns for NSF and other funders, we might be able to reasonably infer presence of an award number based on a pattern (e.g., all NSF award numbers are 7 digit numbers sometimes preceded by 3 letter directorate codes in the form of "1546024", "OPP-1546024", or "OPP 1546024"). That would be pretty reliable for NSF, especially if either the acronym (NSF) or the phrase "National Science Foundation" were in the same text block somewhere.  Patterns would differ for other funders,but even getting a few together would help those documents that provide funding information pass the check.

In addition, given a list of funders such as the CrossRef Funder registry (https://github.com/Crossref/open-funder-registry), we could look for text matches from the list of funder names, acronyms, and identifiers with the text in the `funder` field.  This would then be strong evidence that they passed those checks.  In general, whatever we can do to extract matching info from the text fields would be useful for the check implementation, even if we can't be exhaustively complete.  This could possibly be a different check (that the funder matches one in the funder registry), or maybe it would be part of the existing presence checks depending on how you wanted to handle it.

**Comment by tedhabermann** [comment]
This is awardNumber in DataCite. Other DataCite funder information (with paths):
"fundingReferences": [{
            "awardUri": "data.attributes.fundingReferences.awardUri",
            "awardTitle": "data.attributes.fundingReferences.awardTitle",
            "funderName": "data.attributes.fundingReferences.funderName",
            "awardNumber": "data.attributes.fundingReferences.awardNumber",
            "funderIdentifier": "data.attributes.fundingReferences.funderIdentifier",
            "funderIdentifierType": "data.attributes.fundingReferences.funderIdentifierType",
            "SchemeURI": "data.attributes.fundingReferences.SchemeURI"
        }]
